plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -5365,6 +5365,47 @@ public void testDeleteRetainsTableHistory()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testCreateOrReplaceTable()
+    {
+        String tableName = "test_create_or_replace_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT BIGINT '42' a, DOUBLE '-38.5' b", 1);
+        assertThat(query("SELECT CAST(a AS bigint), b FROM " + tableName))
+                .matches("VALUES (BIGINT '42', -385e-1)");
+
+        long v1SnapshotId = getLatestSnapshotId(tableName);
+
+        assertUpdate("CREATE OR REPLACE TABLE " + tableName + " AS SELECT BIGINT '-42' a, DOUBLE '38.5' b", 1);
+        assertThat(query("SELECT CAST(a AS bigint), b FROM " + tableName))
+                .matches("VALUES (BIGINT '-42', 385e-1)");
+
+        assertThat(query("SELECT a, b  FROM " + tableName + " FOR VERSION AS OF " + v1SnapshotId))
+                .matches("VALUES (BIGINT '42', -385e-1)");
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testCreateOrReplaceTableChangeColumnNamesAndTypes()
+    {
+        String tableName = "test_create_or_replace_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT BIGINT '42' a, DOUBLE '-38.5' b", 1);
+        assertThat(query("SELECT CAST(a AS bigint), b FROM " + tableName))
+                .matches("VALUES (BIGINT '42', -385e-1)");
+
+//        long v1SnapshotId = getLatestSnapshotId(tableName);
+
+        assertUpdate("CREATE OR REPLACE TABLE " + tableName + " AS SELECT VARCHAR 'test' c, VARCHAR 'test2' d", 1);
+        assertThat(query("SELECT c, d FROM " + tableName))
+                .matches("VALUES (VARCHAR 'test', VARCHAR 'test2')");
+
+        // TODO: investigate why this snapshot query fails
+//        assertThat(query("SELECT a, b  FROM " + tableName + " FOR VERSION AS OF " + v1SnapshotId))
+//                .matches("VALUES (BIGINT '42', -385e-1)");
```

Comment:
if something currently fails but it shouldn't, we should have `assertFails` or `assertThrownBy` with a TODO comment and a link.

of course no need to create issues for things that are going to be resolved before something gets merged

Reply:
This has been fixed in the meanwhile.